registry/txt.go

```diff
@@ -123,9 +123,8 @@ func (im *TXTRegistry) Records(ctx context.Context) ([]*endpoint.Endpoint, error
 		if err != nil {
 			return nil, err
 		}
-		key := fmt.Sprintf("%s::%s", im.mapper.toEndpointName(record.DNSName), record.SetIdentifier)
-		labelMap[key] = labels
-		txtRecordsMap[record.DNSName] = struct{}{}
+		labelMap[fmt.Sprintf("%s::%s", im.mapper.toEndpointName(record.DNSName), record.SetIdentifier)] = labels
+		txtRecordsMap[fmt.Sprintf("%s::%s", record.DNSName, record.SetIdentifier)] = struct{}{}
```

Comment:
What do you think about a small function which return's the `Endpoint`'s unique name? Or maybe even a new method in `Endpoint` struct?